### PR TITLE
Warn if volblocksize is smaller than ashift

### PR DESCRIPTION
**Pull request by lundman**
Users easily created ZVOL with a smaller volblocksize than ashift, then find it
writes much more than expected. I didn't want to stop them from being able to do that, but 
at least warn the users of this consequence.

### Motivation and Context

Related to https://github.com/openzfsonwindows/ZFSin/issues/239 and initial commit
https://github.com/openzfsonwindows/ZFSin/pull/240/commits/ee8e361227f8c9b978a123fff2d6b456a3d07b1e

### Description

Simple warning after saving the ashift value.

### How Has This Been Tested?

Tested on Windows. But the better way to add new code is to have it approved here, then trickle down. Also gives us a chance to massage the message into English that is more clear.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

----

Such a setting results in write amplification and excessive space
wastage.

$ ./zfs create -s -V 3GB -b 2048 tank/vol
Warning: volblocksize (2048) < ashift (12 / 4096)
means all writes are amplified and space wasted.

Signed-off-by: Jorgen Lundman <lundman@lundman.net>

### CODE REVIEW

**Comment by lundman** [comment]
Ah printf differences eh, interesting.

File: lib/libzfs/libzfs_dataset.c [hunk 1]
```diff
@@ -3854,6 +3858,14 @@ zfs_create(libzfs_handle_t *hdl, const char *path, zfs_type_t type,
 			    "size"));
 			return (zfs_error(hdl, EZFS_BADPROP, errbuf));
 		}
+
+		if ((ashift > 0ULL) &&
+		    (blocksize < (1ULL << ashift))) {
+			(void) fprintf(stderr, dgettext(TEXT_DOMAIN, "Warning: "
+			    "volblocksize (%llu) < ashift (%llu / %llu)\n"
+			    "means all writes are amplified and space wasted.\n"),
+			    blocksize, ashift, 1ULL << ashift);
```
Comment (behlendorf) [inline lib/libzfs/libzfs_dataset.c:3867]: We've been explicitly adding a `u_longlong_t` cast to resolve the compiler warnings for the `%llu` types.  Alternately, you could use `zfs_nicebytes` to pretty-print some of values with human readable suffixes.  Either solution seems fine to me.

**Comment by DeHackEd** [comment]
```
# zpool list -o name,ashift dehacked
NAME      ASHIFT
dehacked       0
# zpool set ashift=9 dehacked
# zpool list -o name,ashift dehacked
NAME      ASHIFT
dehacked       9
```
Actual ashift on the one vdev here is 12. The "ashift" property itself means relatively little so that concerns me a bit.

**Comment by ahrens** [comment]
I didn't realize that the `ashift` property is disconnected from the vdevs that are in the pool.  We should really be doing this based on `spa_max_ashift`.

**Comment by lundman** [comment]
`spa_max_ashift` isn't really reachable from userland right?

**Comment by ahrens** [comment]
@lundman I don't think so.  We could expose that as a new (hidden?) pool property.  Or in the zpool config, which is even more hidden, and in keeping with how we expose other pool "internals".  But I'd prefer to move away from that since it's much more fragile and undocumented/undocumentable than properties.

**Pull request closed without merging**